crates/ruff_linter/src/checkers/ast/analyze/statement.rs

```diff
@@ -1560,7 +1560,9 @@ pub(crate) fn statement(stmt: &Stmt, checker: &mut Checker) {
                 pylint::rules::named_expr_without_context(checker, value);
             }
             if checker.enabled(Rule::AsyncioDanglingTask) {
-                ruff::rules::asyncio_dangling_task(checker, value);
+                let mut diagnostics: Vec<Diagnostic> = vec![];
+                ruff::rules::asyncio_dangling_task(checker, value, &mut diagnostics);
+                checker.diagnostics.extend(diagnostics);
```

Comment:
Why this change? This doesn't match the pattern we use to add a diagnostic for other rules.

Reply:
Probably because it _is_ the pattern we use for the deferred scopes diagnostics, since they can't take a `mut` on `Checker` :)

I'd probably prefer changing the rule to return `Option<Diagnostic>` though, and then pushing onto the appropriate vector. That should work fine for both call-sites.

Reply:
:D Don't listen to me listen to Charlie.

Reply:
👍 changed to `Option<Diagnostic>`